src/decoder/stream.rs

```diff
@@ -67,6 +68,44 @@ pub trait EndianReader: Read {
     }
 }
 
+/// Reader that decompresses DEFLATE streams
+pub struct DeflateReader {
+    buffer: io::Cursor<Vec<u8>>,
+    byte_order: ByteOrder,
+}
+
+impl DeflateReader {
+    pub fn new<R: Read + Seek>(
+        reader: &mut SmartReader<R>,
+    ) -> io::Result<(usize, Self)> {
```

Comment:
`io::Error` is suboptimal here, prefer a `TiffError` here. Then you can differentiate between `reader` failing and and the `inflate` failing. The latter should have proper diagnosis with a new variant in `TiffFormatError` since I don't think any of the existing ones fit. The error variant should have an opaque type encapsulating the details from `miniz_oxide`.

Reply:
Sounds good, would it be inappropriate to just add a variant named InflateError encapsulating the TINFLStatus miniz returns on error into TiffFormatError? Or are you asking for a more generalized custom enum that would hold constant even if we switched to a different DEFLATE implementation later?

Reply:
Something where the interface doesn't change if `miniz_oxide` is replaced would be optimal. However, the internal representation need not be converted which would optimally make the `Display` and `Debug` implementation informative.

```
enum TiffFormatError {
    InflateError(InflateError),
    …
}

pub struct InflateError {
    status: TINFLStatus,
}
```

Reply:
Actually, `TINFLStatus` doesn't add a `Display` implementation so it may be better to not print the details for display.

Reply:
Yeah I guess it could just have the display implementation be "Failed to decode INFLATE data" or something

Reply:
No all-caps, best to think of display implementations as a standard sentence with Rust capitalization rules.